contrib/jemalloc-cmake/CMakeLists.txt

```diff
@@ -161,6 +161,9 @@ target_include_directories(_jemalloc SYSTEM PRIVATE
 
 target_compile_definitions(_jemalloc PRIVATE -DJEMALLOC_NO_PRIVATE_NAMESPACE)
 
+# Because our coverage callbacks call malloc, and recursive call of malloc could not work.
```

Comment:
This breaks compiler cache of these files:

```
+ sccache --show-stats
Compile requests                   4101
Compile requests executed          4035
Cache hits                         4030
Cache hits (C/C++)                 4030
Cache misses                          5
Cache misses (C/C++)                  5
Cache timeouts                        0
Cache read errors                     0
Forced recaches                       0
Cache write errors                    0
Compilation failures                  0
Cache errors                          0
Non-cacheable compilations            0
Non-cacheable calls                  66
Non-compilation calls                 0
Unsupported compiler calls            0
Average cache write               0.051 s
Average compiler                  7.634 s
Average cache read hit            0.063 s
Failed distributed compilations       0

Non-cacheable reasons:
-fno-profile-instr-generate          66

Cache location                  s3, name: clickhouse-builds, prefix: /ccache/sccache/
Version (client)                0.5.4
```

We should only add this flags to the compilation if `SANITIZE_COVERAGE` is enabled, and leave them empty otherwise.